OPUNetGameSelectWnd.cpp

```diff
@@ -642,21 +644,23 @@ void OPUNetGameSelectWnd::OnReceive(Packet &packet)
 			externalPort = ntohs(packet.tlMessage.echoExternalAddress.addr.sin_port);
 		}
 
-		// Build new net info text string
-		char text[128];
-		_snprintf_s(text, sizeof(text), "External IP: %d.%d.%d.%d:%d", (externalIp.s_addr & 255), (externalIp.s_addr >> 8 & 255), (externalIp.s_addr >> 16 & 255), (externalIp.s_addr >> 24 & 255), externalPort);
-		// Check if internal address received
-		if (bReceivedInternal)
+		// Confine scope of `std::string text` to within a single switch case
 		{
-			// Not quite true, since internal port might be random (no hosting, but possibly still open)
-			//strncat(text, " (Direct Host Capable)", sizeof(text));
-		}
-		if (bTwoExternal)
-		{
-			strncat(text, "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.", sizeof(text));
+			// Build new net info text string
+			std::string text = "External IP: " + FormatIP4Address(externalIp.s_addr) + ":" + std::to_string(externalPort);
+			// Check if internal address received
+			if (bReceivedInternal)
+			{
+				// Not quite true, since internal port might be random (no hosting, but possibly still open)
+				//text += " (Direct Host Capable)";
+			}
+			if (bTwoExternal)
+			{
+				text += "\nWarning: Address and Port-Dependent Mapping detected\nYou may have difficulty joining games.";
+			}
+			// Update net info text
+			SendDlgItemMessage(this->hWnd, IDC_NetInfo, WM_SETTEXT, 0, (long)text.c_str());

```

Comment:
Is there a way to update to a non c-style cast? I don't think static_cast or dynamic_cast would work.

Reply:
For sure we can use C++ style casts. I'll investigate in a future branch. There are a lot of similar C-style casts. I figure I should try to handle them all at once.